src/common/promise.ts

```diff
@@ -0,0 +1,10 @@
+import * as plimit from 'p-limit';
+
+export const runPromises = async <T>(
+  promises: Array<Promise<T>>,
+  concurrency = 10
```

Comment:
This doesn't do what you think it does. Since all the promises have been created they are all already in progress and the concurrency limit is meaningless.

Reply:
Removed p-limit usages